Performance.md review

reekamaharaj opened the pull request:
I have finished initial edits to `Performance.md`. Please review them when you get a chance. 

What I did: I cleaned up some of the language and then changed some formatting around the tables to make them collapsible. Right now they are all defaulted to `open` but they can be changed to default closed if we ever wanted that.

Immediate feedback needed: The section under **Performance Tables** has two options for how that can look. We can keep the bullet points or I can change them to the paragraphs. I think the paragraphs are better because the bullet points are a little confusing, it's a mixture of information and not one clear idea.

Just a thought: The last comment I have would be that we could group the information better. If we split the sections up into "Lurk", "RISC", and "zkLLVM" that makes more sense to me. It would also mean the tables are spread out through the file more. I didn't make any of these edits. If we decide this will be part of the content for the website then I think we should discuss this more.

xc93 commented on Performance.md:
```diff
@@ -1,208 +1,212 @@
 
-The reference pseudocode of these examples are available at the end
-of this document.
+The pseudocode for these examples is provided at the end of this document.
```
Let's delete this sentence as the code is not at the end of the doc.

xc93 commented on Performance.md:
```diff
@@ -1,208 +1,212 @@
+- RISC Zero has its own performance counter, which is used to measure the execution time and cycles.
+- The "CPU" prefix denotes implementation **without** GPU acceleration
+- The "GPU" prefix denotes implementation **with** GPU acceleration
+- Last Update: Dec 19th, 2023
```
There are many parts in this benchmark set and they have different "Last Update" dates. Either we change the date here to be the latest update date, or we simply remove it from here.

xc93 commented on Performance.md:
```diff
@@ -1,208 +1,212 @@
+
+We've also noted where the implementation is using CPU or GPU acceleration. If you see the prefix "CPU", this refers to an implementation without GPU acceleration, whereas "GPU" denotes an implementation that utilized GPU acceleration.
+
+This section was last updated Dec 19th, 2023.
```
I do like the paragraph-version more. Let's use it.

xc93 reviewed (changes requested):
I like these edits very much! Great work! Some minor comments can be found in my review. 

We can group the page by the ZK backend (Cairo, Lurk, ...), and for each such subsection (of a given ZK backend) we list the numbers for Direct Implementation, Proof of Proof, and Implementation Details. That sounds good to me. We can implement that change in next PR.

reekamaharaj commented:
@xc93 I have updated it to remove the sentence about the pseudocode, changed the section on performance to the paragraph version, and removed the "last updated" dates.

I will get started on rearranging everything to be grouped by ZKbackends. We can get this merged in and I will submit another PR for the restructuring, that will take me a little more time to make sure it makes sense.

xc93 reviewed (approved):
LGTM.
